tests/unit/ui/components/TestConfirmButton.js

```diff
@@ -99,121 +129,33 @@ describe(__filename, () => {
-      'puffy',
-      false,
-    );
-    expect(root.find('.ConfirmButton-confirm-button').children()).toHaveText(
-      confirmButtonText,
-    );
-  });
-
-  it('passes props to the cancel button in the confirmation panel', () => {
-    const { store } = dispatchClientMetadata();
-    const cancelButtonText = 'neutral cancel button';
-    const cancelButtonType = 'neutral';
-    const root = render({ cancelButtonText, cancelButtonType, store });
-
-    // Show the confirmation panel.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find('.ConfirmButton-cancel-button')).toHaveProp(
-      'buttonType',
-      cancelButtonType,
-    );
-    expect(root.find('.ConfirmButton-cancel-button')).toHaveProp(
-      'puffy',
-      false,
-    );
-    expect(root.find('.ConfirmButton-cancel-button').children()).toHaveText(
-      cancelButtonText,
-    );
-  });
-
-  it('can make the buttons in the confirmation panel puffy', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ puffyButtons: true, store });
-
-    // Show the confirmation panel.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find('.ConfirmButton-cancel-button')).toHaveProp('puffy', true);
-    expect(root.find('.ConfirmButton-confirm-button')).toHaveProp(
-      'puffy',
-      true,
-    );
-  });
-
-  it('can render a confirmation without a message', () => {
-    const { store } = dispatchClientMetadata();
-    const root = render({ message: undefined, store });
-
-    // Show the confirmation panel.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root.find('.ConfirmButton-message')).toHaveLength(0);
-  });
-
-  it('closes the confirmation panel on cancel ', () => {
+  it('hides ConfirmationDialog via onCancel', () => {
     const { store } = dispatchClientMetadata();
-    const onConfirmSpy = sinon.spy();
-    const root = render({ onConfirm: onConfirmSpy, store });
-
-    expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
+    const root = renderWithDialog({ store });
 
-    // Show the confirmation panel.
-    root.find(Button).simulate('click', createFakeEvent());
-    applyUIStateChanges({ root, store });
-
-    expect(root).toHaveClassName('ConfirmButton--show-confirmation');
-    sinon.assert.notCalled(onConfirmSpy);
+    const onCancel = getDialogProp(root, 'onCancel');
+    // Simulate a cancellation.
+    onCancel(createFakeEvent());
 
-    root
-      .find('.ConfirmButton-cancel-button')
-      .simulate('click', createFakeEvent());
     applyUIStateChanges({ root, store });
-
     expect(root).not.toHaveClassName('ConfirmButton--show-confirmation');
-    sinon.assert.notCalled(onConfirmSpy);
+    expect(root.find(ConfirmationDialog)).toHaveLength(0);
```

Comment:
should we show above that the dialog existed prior to clicking on Cancel:
 `expect(root.find(ConfirmationDialog)).toHaveLength(1);`

Reply:
Sure, I can add that. It's a kind of a duplication of other test code but since we don't have Flow coverage in tests, mistakes can happen 🤷‍♀️